django_project/frontend/utils/parcel.py

```diff
@@ -33,26 +34,41 @@ def find_parcel_base(cls, serialize_cls,
                      other: GEOSGeometry, parcel_keys=[]):
     """Base function to find parcel."""
     used_parcels = []
-    parcels = cls.objects.filter(geom__intersects=other)
+    results = []
+    cname_list = []
+    parcels = cls.objects.filter(geom__bboverlaps=other)
     if parcel_keys:
         parcels = parcels.exclude(
             cname__in=parcel_keys
         )
-    parcels = parcels.order_by('cname').distinct('cname')
-    if parcels:
-        cname_list = [a.cname for a in parcels]
+    parcels = parcels.annotate(
+        centroid=Centroid('geom')).order_by('cname').distinct('cname')
+    if parcels.exists():
```

Comment:
@danangmassandy, the code is nested too deeply. Let's refactor it to make it more readable

Reply:
Updated Dimas,,